src/main/java/com/google/sps/servlets/StorePromotionQuestion.java

```diff
@@ -0,0 +1,116 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.QuestionChoice;
+import com.google.sps.data.QuestionDatabase;
+import com.google.sps.data.QuizQuestion;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.PatternSyntaxException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Stores career question data in database */
+@WebServlet("/storePromotion")
+public class StorePromotionQuestion extends HttpServlet {
+  private static final String JSON_CONTENT_TYPE = "application/json";
+  private static final String FORM_SUBMIT_PARAMETER = "promotionQuizSubmit";
+  private static final String QUESTION_PARAMETER = "promotionQuestion";
+  private static final String CHOICETEXT_PARAMETER = "promotionChoiceText";
+  private static final String CHOICE_IS_ACCEPTED_PARAMETER = "promotionAcceptedChoice";
+  private static final String QUIZ_CAREERPATH_PARAMETER = "promotionCareerPath";
+  private static final String QUIZ_LEVEL_PARAMETER = "promotionQuizLevel";
+  private static final String ACCEPTED_CHOICE = "y";
+  private static final String NOT_ACCEPTED_CHOICE = "n";
+  private static final String CANT_SPLIT_STRING =
+      "Choices and whether they are accepted need to be split by semicolons.";
+  private static final String NUMBER_CHOICES_DO_NOT_MATCH_ACCEPTED_SPEC =
+      "The specification of the choice options of being accepted or not is the same length as the "
+          + "number of choices.";
+  private static final String IS_ACCEPTED_INCORRECT_FORMAT =
+      "The specification of the choice options of being accepted or not is not given by y or n.";
+  private static final String REDIRECTION_URL = "admin/AddStaticData.html";
+  private static Gson gson;
+  private DatastoreService datastore;
+
+  @Override
+  public void init() {
+    this.gson = new Gson();
+    this.datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    if (request.getParameter(FORM_SUBMIT_PARAMETER) != null) {
+      addPromotionQuizQuestion(request, response);
+    }
```

Comment:
What happens if it is null?

Reply:
If the form button is not clicked (null), then nothing happens. I believe this is the expected behavior as nothing should be added to the database.